enterprise/models.py

```diff
@@ -94,21 +109,33 @@ class EnterpriseCustomerUser(TimeStampedModel):
     """
 
     enterprise_customer = models.ForeignKey(EnterpriseCustomer, blank=False, null=False)
-    user_id = models.PositiveIntegerField(null=False, blank=False)
```

Comment:
@e-kolpakov You can also move a lot more of the new `django.contrib.auth.models.User` references to the `admin.py` code rather than `models.py` if you want to keep the models code more "pure". I think the only reference you need in `models.py` is within `@property def user(self):` and even that could technically be removed if other parts of the code can live without that convenience method. The `post_save` hook might need to be moved to another file, because if a non-LMS app starts using this, we wouldn't want it registering that signal handler (assuming user_id is always assumed to be an LMS user and not "user in the current IDA").

@mattdrayer Has any work around that sort of user data service been done, for the edX IDA architecture in general?